File: ui/src/services/post-service/model/PostDTO.ts
import type {PostReactionDTO} from "src/services/post-service/model/PostReactionDTO";
import type {UserDTO} from "@/services/user-service/model/UserDTO";
import type {CommentDTO} from "@/services/comment-service/model/CommentDTO";

export interface PostDTO {
    uuid: string;
    content: string;
    createdAt: Date;
    user: UserDTO;
    images: string[];
    comments: CommentDTO[];
    postReactions: PostReactionDTO[];
}

File: src/main/java/socialmediaapp/data_services/post/model/PostDTO.java
package socialmediaapp.data_services.post.model;

import lombok.*;
import socialmediaapp.data_services.comment.model.CommentDTO;
import socialmediaapp.data_services.user.model.UserDTO;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor

public class PostDTO {
    private String uuid;
    private String content;
    private Date createdAt;
    private UserDTO user;

    private List<byte[]> images;
    private List<CommentDTO> comments;
    private List<PostReactionDTO> postReactions;
}
